src/service/mercury/index.ts

```diff
@@ -154,33 +154,29 @@ export class MercuryClient {
 
   renewAndRetry = async <T>(
     method: () => Promise<T>,
-    network: NetworkNames
-  ) => {
+    network: NetworkNames,
+    retryCount?: number
+  ): Promise<T> => {
     try {
       return await method();
     } catch (error: any) {
       // renew and retry 0n 401, otherwise throw the error back up to the caller
-      if (error instanceof Error) {
-        if (error.message.includes(ERROR_MESSAGES.JWT_EXPIRED)) {
-          await this.renewMercuryToken(network);
-          this.logger.info("renewed expired jwt");
-          return await method();
-        }
-
-        this.logger.error(error.message);
-        throw new Error(error.message);
-      }
-
-      // In the case of subscription posts or non graphQL queries, the 401 response is different
-      if (error.response?.status === 401) {
+      if (
+        error.message?.includes(ERROR_MESSAGES.JWT_EXPIRED) ||
+        error.response?.status === 401
+      ) {
         await this.renewMercuryToken(network);
         this.logger.info("renewed expired jwt");
         return await method();
       }
 
-      const _error = JSON.stringify(error);
-      this.logger.error(_error);
-      throw new Error(_error);
+      // Retry in non 401 cases
+      if (retryCount) {
```

Comment:
I thought about but decided against a backoff here because this will happen in the request lifecycle and is meant to handle transient errors so it felt like slowing this down here wasn't the right move.